examples/slice/slice_mountain_paradiag.py

```diff
@@ -0,0 +1,222 @@
+                cp=cp, R_d=R_d, p_0=p_0, kappa=kappa, g=g, Up=Up,
+                top=True, Pi=Pi)
+p1 = maximum(Pi)
+alpha = 2.*(p1-p0)
+beta = p1-alpha
+pi_top = (1.-beta)/alpha
+
+hydrostatic_rho(Vv, V2, mesh, thetan, rhon, pi_boundary=fd.Constant(pi_top),
+                cp=cp, R_d=R_d, p_0=p_0, kappa=kappa, g=g, Up=Up,
+                top=True)
+
+rho_back = fd.Function(V2).assign(rhon)
+
+zc = H-10000.
+mubar = 0.3
+mu_top = fd.conditional(z <= zc, 0.0, mubar*fd.sin((np.pi/2.)*(z-zc)/(H-zc))**2)
+mu = fd.Function(V2).interpolate(mu_top/dT)
+
+form_function = get_form_function(n, Up, c_pen=2.0**(-7./2),
+                                  cp=cp, g=g, R_d=R_d,
+                                  p_0=p_0, kappa=kappa, mu=mu)
+
+form_mass = get_form_mass()
+
+zv = fd.as_vector([fd.Constant(0.), fd.Constant(0.)])
+bcs = [fd.DirichletBC(W.sub(0), zv, "bottom"),
+       fd.DirichletBC(W.sub(0), zv, "top")]
+
+for bc in bcs:
+    bc.apply(Un)
+
+# Parameters for the diag
+lines_parameters = {
+    "ksp_type": "gmres",
+    "ksp_converged_reason": None,
+    "pc_type": "python",
+    "pc_python_type": "firedrake.AssembledPC",
+    "assembled_pc_type": "python",
+    "assembled_pc_python_type": "firedrake.ASMVankaPC",
+    "assembled_pc_vanka_construct_dim": 0,
+    "assembled_pc_vanka_sub_sub_pc_type": "lu",
+    "assembled_pc_vanka_sub_sub_pc_factor_mat_solver_type": 'mumps',
+}
+
+solver_parameters_diag = {
+    "snes_lag_preconditioner_persists": None,
+    "snes_lag_preconditioner": 4,
+    "ksp_type": "fgmres",
+    "ksp_monitor": None,
+    "ksp_converged_reason": None,
+    "ksp_atol": 1e-8,
+    "ksp_rtol": 1e-8,
+    "ksp_max_it": 400,
+    'snes_monitor': None,
+    'snes_converged_reason': None,
+    'mat_type': 'matfree',
+    'pc_type': 'python',
+    'pc_python_type': 'asQ.DiagFFTPC'}
+
+for i in range(np.sum(M)):
+    solver_parameters_diag["diagfft_"+str(i)+"_"] = lines_parameters
+
+dt = 5
+dT.assign(dt)
+
+t = 0.
+
+alpha = 1.0e-4
+theta = 0.5
+
+PD = asQ.paradiag(ensemble=ensemble,
+                  form_function=form_function,
+                  form_mass=form_mass, w0=Un,
+                  dt=dt, theta=theta,
+                  alpha=alpha, time_partition=M, bcs=bcs,
+                  solver_parameters=solver_parameters_diag,
+                  circ="none",
+                  tol=1.0e-6, maxits=None,
+                  ctx={}, block_mat_type="aij")
+
+r = PD.time_rank
+
+# only last slice does diagnostics/output
+if PD.time_rank == len(M)-1:
+
+    uout = fd.Function(V1, name='velocity')
+    thetaout = fd.Function(Vt, name='temperature')
+    rhoout = fd.Function(V2, name='density')
+
+    ofile = fd.File('slice_mountain_diag.pvd',
+                    comm=ensemble.comm)
+
+    def assign_out_functions():
+        uout.assign(PD.aaos.w_all.split()[-3])
+        rhoout.assign(PD.aaos.w_all.split()[-2] - rho_back)
+        thetaout.assign(PD.aaos.w_all.split()[-1] - theta_back)
```

Comment:
These three lines can use `aaos.get_component(-1, cpt)` now instead of indexing into `w_all` directly